### PR TITLE
ci: update xapi-project.github.io on ever commit

### PR DESCRIPTION
**Pull request by psafont**
This replaces the previous mechanism using a shell script. It hadn't been run in quite a while!

Signed-off-by: Pau Ruiz Safont <pau.safont@citrix.com>

### CODE REVIEW

**Review by edwintorok — APPROVED** [review]
Was this previously done by hand?
Can we test this in any other way than by merging this commit to master?

**Comment by psafont** [comment]
This use to be automated mid-2019, when it broke without anybody noticing. Tina did a few manually, but it's been mainly forgotten.

Unfortunately the only way to test this is by merging to master, if something goes wrong we can restore the commit / repository (I have an updated local copy of it)

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -53,3 +54,19 @@ jobs:
           destination_dir: source/includes/
           allow_empty_commit: false
           enable_jekyll: true # do not create .nojekyll file
+
+      - name: Generate xapi docs
+        run: make doc-json
+
+      - name: Deploy xapi docs
+        uses: peaceiris/actions-gh-pages@v3
+        with:
```
Comment (robhoes) [inline .github/workflows/docs.yml:63]: Where does it say which files to copy from what is generated by `make doc`?

Comment (robhoes) [reply .github/workflows/docs.yml:63]: The previous script wrote files like this:
```
cp -f $HOME/classes/* xen-api/classes/
cp -f $HOME/releases/* xen-api/releases/
cp -f $HOME/xenapi.json _data/
cp -f $HOME/releases.yml _data/
cp -f $HOME/release_info.json _data/
```

Comment (psafont) [reply .github/workflows/docs.yml:63]: Tina changed where the files are placed when `make gen-json` was added to have the proper structure: 6c26f17984b4e3561c9b851560f4d31d6a15aa8e

The rule takes all the generated files in `.../jekyll/`, like [`make install`](https://github.com/xapi-project/xen-api/blob/master/Makefile#L198) (defined here by `publish_dir: ${{ env.JEKYLL_DOCDIR }}`) and puts them in the root of the docs repository. Only the paths specifies by the previous script have files, see the outputs of doc-json when run locally:
```
$ tree _build/install/default/xapi/doc/jekyll/ 
_build/install/default/xapi/doc/jekyll/
├── _data
│   ├── release_info.json
│   ├── releases.yml
│   └── xenapi.json
└── xen-api
    ├── classes
    │   ├── auth.md
    │   ├── blob.md
    │   ├── bond.md
    │   ├── certificate.md
    │   ├── cluster_host.md
    │   ├── cluster.md
    │   ├── console.md
    │   ├── crashdump.md
    │   ├── data_source.md
    │   ├── dr_task.md
    │   ├── event.md
    │   ├── feature.md
    │   ├── gpu_group.md
    │   ├── host_cpu.md
    │   ├── host_crashdump.md
    │   ├── host.md
    │   ├── host_metrics.md
    │   ├── host_patch.md
    │   ├── lvhd.md
    │   ├── message.md
    │   ├── network.md
    │   ├── network_sriov.md
    │   ├── pbd.md
    │   ├── pci.md
    │   ├── pgpu.md
    │   ├── pif.md
    │   ├── pif_metrics.md
    │   ├── pool.md
    │   ├── pool_patch.md
    │   ├── pool_update.md
    │   ├── probe_result.md
    │   ├── pusb.md
    │   ├── pvs_cache_storage.md
    │   ├── pvs_proxy.md
    │   ├── pvs_server.md
    │   ├── pvs_site.md
    │   ├── repository.md
    │   ├── role.md
    │   ├── sdn_controller.md
    │   ├── secret.md
    │   ├── session.md
    │   ├── sm.md
    │   ├── sr.md
    │   ├── sr_stat.md
    │   ├── subject.md
    │   ├── task.md
    │   ├── tunnel.md
    │   ├── usb_group.md
    │   ├── user.md
    │   ├── vbd.md
    │   ├── vbd_metrics.md
    │   ├── vdi.md
    │   ├── vdi_nbd_server_info.md
    │   ├── vgpu.md
    │   ├── vgpu_type.md
    │   ├── vif.md
    │   ├── vif_metrics.md
    │   ├── vlan.md
    │   ├── vm_appliance.md
    │   ├── vm_guest_metrics.md
    │   ├── vm.md
    │   ├── vm_metrics.md
    │   ├── vmpp.md
    │   ├── vmss.md
    │   ├── vtpm.md
    │   └── vusb.md
    └── releases
        ├── boston.md
        ├── clearwater-felton.md
        ├── clearwater.md
        ├── clearwater-whetstone.md
        ├── cowley.md
        ├── cream.md
        ├── creedence.md
        ├── dundee.md
        ├── ely.md
        ├── falcon.md
        ├── george.md
        ├── indigo.md
        ├── inverness.md
        ├── jura.md
        ├── kolkata.md
        ├── lima.md
        ├── miami.md
        ├── midnight-ride.md
        ├── naples.md
        ├── next.md
        ├── orlando.md
        ├── orlando-update-1.md
        ├── quebec.md
        ├── rio.md
        ├── stockholm.md
        ├── stockholm_psr.md
        ├── symc.md
        ├── tampa.md
        ├── vgpu-productisation.md
        └── vgpu-tech-preview.md

4 directories, 99 files
```

**Review by robhoes — APPROVED** [review]

**Pull request merged**